tests/optim/test_oss.py

```diff
@@ -262,6 +262,9 @@ def run_test_zero_grad(rank, world_size, tempfile_name):
 
 def test_zero_grad():
     world_size = 2
+    if torch.cuda.is_available() and torch.cuda.device_count() < world_size:
+        world_size = min(world_size, torch.cuda.device_count())
```

Comment:
can world_size be 1 allowed here?

Reply:
this comes from another branch actually, it was just to check CI. Doing a mega-refactor of this ugly piece of code (my bad) right now